test/gorest-api/gorest-api.test.js

```diff
@@ -12,82 +12,178 @@ describe("GoRest Users API test (/users)", () =>{
       unAuthenticatedApi = new GoRestUsersApi();
 
       api.authenticate();
+      /*api.authenticate({username: "Test", password: "12345"
+      });*/
     })
 
-    it("Can get a list of users (GET /users", async () => {
-      const response = await api.getUsersList();
-      console.log('\x1b[36m%s\x1b[0m', JSON.stringify(response, null, 2));
+  describe("GoRest Users API miscellaneous tests", () => {
+      it("Error message returned when creating a user as an authenticated user (POST /users", async () => {
+        const userToBeCreated = {
+          "name":"Tenali Ramakrishna", 
+          "gender":"male",
+          "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`, 
+          "status":"active"
+        }
+  
+        const err = await guard(async () => unAuthenticatedApi.createUser(userToBeCreated));
+              
+        expect(err).to.have.property("statusCode", 401);
+        expect(err).to.be.an.instanceOf(Error);
+        expect(err).to.have.property("message", "401 - {\"meta\":null,\"data\":{\"message\":\"Authentication failed\"}}");
+      })
 
-      expect(response.meta.pagination).to.be.an("object");
-      expect(response.data, "Expected data array to have 20 results").to.be.an("array").that.has.lengthOf(20);
-      expect(response.meta.pagination).to.have.property("page", 1)
-    })
+      it("Error returned when creating a user and required user properties are missing from the request (POST /public/v1/users", async () => {
+        const userToBeCreated = {
+          "name":"Tenali Ramakrishna", 
+          "gender":"male",
+          "email": `tenali.ramakrishna-${Math.random().toString(36).slice(2)}@15ce.com`        
+        }
 
+        const err1 = await guard(async () => api.createUser(userToBeCreated));        
+        
+        expect(err1).to.have.property("statusCode", 422);
```

Comment:
No need to differentiate between variable names when they are in different functions. I'll explain.